src/commands/test/test_environment_exceptions.py

```diff
@@ -10,6 +11,28 @@ class MissingExceptException(ReportedException):
     pass
 
 
+class ExceptMissmatchException(ReportedException):
+    def __init__(
+        self,
+        expected_name: Union[str, None],
+        expected_message: Union[str, None],
+        received: StarkException,
+    ):
+        self.expected_name = expected_name
+        self.expected_message = expected_message
+        self.received = received
+        super().__init__()
+
+    def __str__(self) -> str:
+        message = [
+            "Expected:",
+            f"name: {self.expected_name}, message: {self.expected_message}",
+            "Instead got:",
+            f"name: {self.received.code.name}, message: {self.received.message}",
```

Comment:
We should consider printing the message in a new line. Usually, the message is very long.